cli/command/container/stop.go

```diff
@@ -39,16 +40,45 @@ func NewStopCommand(dockerCli *command.DockerCli) *cobra.Command {
 
 func runStop(dockerCli *command.DockerCli, opts *stopOptions) error {
 	ctx := context.Background()
+	timeout := time.Duration(opts.time) * time.Second
 
+	var waitAll sync.WaitGroup
 	var errs []string
+	const maxParallel int32 = 50
+
+	// make sure result is printed in order
+	output := map[string]chan error{}
 	for _, container := range opts.containers {
-		timeout := time.Duration(opts.time) * time.Second
-		if err := dockerCli.Client().ContainerStop(ctx, container, &timeout); err != nil {
-			errs = append(errs, err.Error())
-		} else {
-			fmt.Fprintf(dockerCli.Out(), "%s\n", container)
+		output[container] = make(chan error, 1)
+	}
+	waitAll.Add(1)
+	go func() {
+		for _, container := range opts.containers {
+			if err := <-output[container]; err != nil {
+				errs = append(errs, err.Error())
+			} else {
+				fmt.Fprintf(dockerCli.Out(), "%s\n", container)
+			}
 		}
+		waitAll.Done()
+	}()
```

Comment:
Why does this need to be in a go-router?

Wouldn't it be easier to just do this printing at the end after `waitAll.Wait()` ? Do we gain anything by doing it in a go-routine? Printing to stdout should be pretty fast, so I would think we gain very little by doing this in parallel with the stop tasks. Plus you might have to wait for them all to finish if the first container id is the last to finish, so you wouldn't really save any time.